README


This archive contains the distribution Dist-Zilla-Plugin-Git-Contributors,
version 0.008:

  Add contributor names from git to your distribution

This software is copyright (c) 2014 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.021.



Changes
Revision history for Dist-Zilla-Plugin-Git-Contributors

0.008     2014-10-22 03:30:59Z
          - fix test that broke with changes in Dist::Zilla::Tester 5.021 (it
            was my own patch, even!)

0.007     2014-08-16 06:31:41Z
          - added the new "path" configuration option, for help in building
            dists contained within a subdirectory of a repository

0.006     2014-08-15 01:44:00Z
          - increase the minimum required version of Unicode::Collate
          - increase the minimum required version of Git::Wrapper
          - avoid doubly-encoded committer strings on older perls by avoiding
            encoding the string before we send it to IPC::Open3
          - make sure not to crash doing "dzil build" when a user has no
            user.name and include_releaser = 0 (Kent Fredric, #1)

0.005     2014-08-13 00:46:11Z
          - basic tests are now ascii-only; unicode tests are marked as TODO
            on MSWin32 for now (see (github) genehack/Git-Wrapper#52 for the
            gory details)
          - now properly sorting names using case-insensitive unicode collation

0.004     2014-08-08 01:29:08Z
          - when on MSWin32, check for mysysgit version and also set
            appropriate config options for proper unicode support
          - new "order_by" configuration option, to allow sorting contributors
            by number of commits made, rather than alphabetically
          - mark "include_releaser" option as experimental, as it will likely
            get removed if CPAN-API/metacpan-web#1270 gets resolved

0.003     2014-08-07 02:32:19Z
          - new 'include_releaser' option, for filtering the releaser user out
            of the contributor list after 'include_authors' has been applied
            (include_authors = 1, include_releaser = 0 recommended)

0.002     2014-08-06 05:21:51Z
          - do not add an empty x_contributors key if there are no
            (non-author) contributors to report
          - fix test failures when run on a system where git has not been
            configured
          - warn at runtime if Pod::Weaver::Section::Contributors is too old
            to pull data from distmeta on its own
          - do not blow up if run in a directory that is not tracked by git

0.001     2014-07-31 01:42:08Z
          - Initial release.


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Git-Contributors was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.008;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Add contributor names from git to your distribution",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Git-Contributors",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::Git::Contributors",
  "PREREQ_PM" => {
    "Data::Dumper" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Git::Wrapper" => "0.035",
    "List::Util" => "1.33",
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Path::Tiny" => 0,
    "Safe::Isa" => 0,
    "Try::Tiny" => 0,
    "Unicode::Collate" => "0.53",
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Exporter" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Sort::Versions" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "Test::Requires" => 0,
    "if" => 0,
    "lib" => 0,
    "parent" => 0,
    "utf8" => 0
  },
  "VERSION" => "0.008",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Data::Dumper" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Git::Wrapper" => "0.035",
  "List::Util" => "1.33",
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Path::Tiny" => 0,
  "Safe::Isa" => 0,
  "Sort::Versions" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "Test::Requires" => 0,
  "Try::Tiny" => 0,
  "Unicode::Collate" => "0.53",
  "if" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "parent" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-Git-Contributors was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.009
use strict;
use warnings;

if (eval 'use Module::Build::Tiny 0.039; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Git-Contributors was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.008;
    # use Module::Build::Tiny 0.039;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.021.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.039"
      },
      "dist_abstract" => "Add contributor names from git to your distribution",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Git-Contributors",
      "dist_version" => "0.008",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Git::Contributors",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Data::Dumper" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::MetaProvider" => 0,
        "Git::Wrapper" => "0.035",
        "List::Util" => "1.33",
        "Moose" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "Path::Tiny" => 0,
        "Safe::Isa" => 0,
        "Try::Tiny" => 0,
        "Unicode::Collate" => "0.53",
        "namespace::autoclean" => 0,
        "perl" => "5.008",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Exporter" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Sort::Versions" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "Test::Requires" => 0,
        "if" => 0,
        "lib" => 0,
        "parent" => 0,
        "utf8" => 0
      }
    );


    my %fallback_build_requires = (
      "Exporter" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Sort::Versions" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "Test::Requires" => 0,
      "if" => 0,
      "lib" => 0,
      "parent" => 0,
      "utf8" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
